packages/server/lib/browsers/webkit.ts

```diff
@@ -28,7 +29,26 @@ export function connectToExisting () {
   throw new Error('Cypress-in-Cypress is not supported for WebKit.')
 }
 
+/**
+ * Playwright adds an `exit` event listener to run a cleanup process. It tries to use the current binary to run a Node script by passing it as argv[1].
+ * However, the Electron binary does not support an entrypoint, leading Cypress to think it's being opened in global mode (no args) when this fn is called.
+ * Solution is to filter out the problematic function.
+ * TODO(webkit): do we want to run this cleanup script another way?
+ * @see https://github.com/microsoft/playwright/blob/7e2aec7454f596af452b51a2866e86370291ac8b/packages/playwright-core/src/utils/processLauncher.ts#L191-L203
+ */
+function removeBadExitListener () {
+  const killProcessAndCleanup = process.rawListeners('exit').find((fn) => fn.name === 'killProcessAndCleanup')
+
+  // @ts-expect-error Electron's Process types override those of @types/node, leading to `exit` not being recognized as an event
+  if (killProcessAndCleanup) process.removeListener('exit', killProcessAndCleanup)
+  else debug('did not find killProcessAndCleanup, which may cause interactive mode to unexpectedly open')
+}
+
 export async function open (browser: Browser, url: string, options: BrowserLaunchOpts, automation: Automation): Promise<BrowserInstance> {
+  if (!options.experimentalWebKitSupport) {
+    throw new Error('WebKit was launched, but the experimental feature was not enabled. Please add `experimentalWebKitSupport: true` to your config file to launch WebKit.')
```

Comment:
This is reached if the user passes `--testingType --browser webkit` and bypasses the GUI.